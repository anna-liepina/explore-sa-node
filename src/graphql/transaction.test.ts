//@ts-nocheck
describe('GraphQL: Transaction', () => {
    describe('transactionSearch [presets: perPage 3]', () => {
        it(`without criteria`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(perPage: 3) {
                        id
                        price
                        date
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`within date range`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(from: "2010-01-01", to: "2021-01-01", perPage: 3) {
                        id
                        price
                        date
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`within date range: 2nd page`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(from: "2010-01-01", to: "2021-01-01", perPage: 3, page: 2) {
                        id
                        price
                        date
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        /** there are only 3 results with this postcode, which should fall on 1st page */
        it(`with postcode range: 1st page`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(postcode: "E20 1AB", perPage: 3) {
                        id
                        price
                        date
                        property {
                            street
                            paon
                            saon
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        /** there are only 3 results with this postcode, which should fall on 1st page */
        it(`with postcode range: 2nd page`, async () => {
            const { data: { transactionSearch: data } } = await query({
                query: `
                {
                    transactionSearch(postcode: "E20 1AB", perPage: 3, page: 2) {
                        id
                        price
                        date
                        property {
                            street
                            paon
                            saon
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });
    });
});
