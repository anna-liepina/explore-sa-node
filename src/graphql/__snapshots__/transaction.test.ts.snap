// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] with postcode range: 1st page 1`] = `
[
  {
    "date": "2019-03-25",
    "id": "3",
    "price": 341250,
    "property": {
      "paon": "cc551cd9b04a9cc3da0fd21025e8db42828d5299",
      "saon": "fe78fa628e2a410375cb37144fbdff5a766fcee3",
      "street": "PRIZE WALK",
    },
  },
  {
    "date": "2019-04-25",
    "id": "1",
    "price": 375000,
    "property": {
      "paon": "cc551cd9b04a9cc3da0fd21025e8db42828d5299",
      "saon": "cd435dfdc23e7d95665b678f176329c14ac2af4a",
      "street": "PRIZE WALK",
    },
  },
  {
    "date": "2019-04-26",
    "id": "2",
    "price": 545000,
    "property": {
      "paon": "cc551cd9b04a9cc3da0fd21025e8db42828d5299",
      "saon": "1b762366cdf7cc97e2764e932f60c0edc61b5e86",
      "street": "PRIZE WALK",
    },
  },
]
`;

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] with postcode range: 2nd page 1`] = `[]`;

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] within date range 1`] = `
[
  {
    "date": "2013-10-23",
    "id": "65",
    "price": 100000,
  },
  {
    "date": "2013-12-03",
    "id": "67",
    "price": 130000,
  },
  {
    "date": "2013-12-18",
    "id": "18",
    "price": 125000,
  },
]
`;

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] within date range: 2nd page 1`] = `
[
  {
    "date": "2015-10-15",
    "id": "455",
    "price": 765000,
  },
  {
    "date": "2015-10-30",
    "id": "431",
    "price": 757055,
  },
  {
    "date": "2015-10-30",
    "id": "453",
    "price": 745000,
  },
]
`;

exports[`GraphQL: Transaction transactionSearch [presets: perPage 3] without criteria 1`] = `
[
  {
    "date": "2007-08-22",
    "id": "36",
    "price": 229400,
  },
  {
    "date": "2013-10-23",
    "id": "65",
    "price": 100000,
  },
  {
    "date": "2013-12-03",
    "id": "67",
    "price": 130000,
  },
]
`;
